src/auth/pluggable-auth-client.ts

```diff
@@ -246,14 +262,19 @@ export class PluggableAuthClient extends BaseExternalAccountClient {
       envMap.set('GOOGLE_EXTERNAL_ACCOUNT_TOKEN_TYPE', this.subjectTokenType);
       // Always set to 0 because interactive mode is not supported.
       envMap.set('GOOGLE_EXTERNAL_ACCOUNT_INTERACTIVE', '0');
+      if (this.outputFile) {
+        envMap.set('GOOGLE_EXTERNAL_ACCOUNT_OUTPUT_FILE', this.outputFile);
+      }
       const serviceAccountEmail = this.getServiceAccountEmail();
       if (serviceAccountEmail) {
         envMap.set(
           'GOOGLE_EXTERNAL_ACCOUNT_IMPERSONATED_EMAIL',
           serviceAccountEmail
         );
       }
-      executableResponse = await this.retrieveResponseFromExecutable(envMap);
+      executableResponse = await this.handler.retrieveResponseFromExecutable(
+        envMap
+      );
     }
 
     if (executableResponse) {
```

Comment:
yes, since executableResponse is set to be type ExecutableResponse | undefined (which is needed to have the behavior where retrieveCachedResponse can return undefined if no response is found). If this if check doesn't surround the rest of the logic it will throw ts errors when trying to access the executableResponse properties

Reply:
seems like retrieveResponseFromExecutable either returns it or throws?